pkg/scale/decode.go

```diff
@@ -121,46 +120,94 @@ func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 	switch in.(type) {
 	case *big.Int:
 		err = ds.decodeBigInt(dstv)
+		if err != nil {
```

Comment:
The linter is complaining about ```
pkg/scale/decode.go:118:1: cyclomatic complexity 33 of func `(*decodeState).unmarshal` is high (> 30) (gocyclo)
func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
```
Any suggestion about how to address this is welcome.

Reply:
just add `//nolint:gocyclo` to the line.  We're iterating over the types we need to support, so I don't see any benefit to breaking this up.